TestFoundation/TestNSData.swift

```diff
@@ -4140,5 +4141,19 @@ extension TestNSData {
         XCTAssertEqual(data[data.startIndex.advanced(by: 1)], 0xFF)
 #endif
     }
+
+    func test_nskeyedarchiving() {
+        let bytes: [UInt8] = [0xd, 0xe, 0xa, 0xd, 0xb, 0xe, 0xe, 0xf]
+        let data = NSData(bytes: bytes, length: bytes.count)
+
+        let archiver = NSKeyedArchiver()
+        let key = "data"
+        data.encode(with: archiver)
+        let encodedData = archiver.encodedData
+
+        let unarchiver = NSKeyedUnarchiver(forReadingWithData: encodedData)
```

Comment:
According to https://developer.apple.com/documentation/foundation/nskeyedunarchiver/1410862-init the method is `init(forReadingWith data: Data)` so needs a small adjustment.